### ISSUE TITLE
Autocomplete onSelect callback can't be used with validateFields, as the value haven't been changed

### ISSUE THREAD

**Issue by tylik1** [ant-design/ant-design#12627]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.10.1

### Environment
Google Chrome latest version

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/3q27nzolnp)

### Steps to reproduce
Make sure onSelect callback is attached to Autocomplete module,
Try to call props.form.validateFields on the same Autocomplete module, from onSelect callback
In the demo link, type "ch", then select country from dropdown

### What is expected?
props.form.validateFields receive the same value as onSelect callback

### What is actually happening?
props.form.validateFields receives typed value or undefined if nothing was typed, but not the value from onSelect

---
Once selected autocomplete dropdown option item,  open the console, and you will see 2 outputs:
the one with warning, is the wrong one, while there is also correct one from onSelect callback below.
React also throws error popup, but it is not related, as on my dev environment, i don't see it.

**Comment by tylik1** [comment]
setTiemout with 250ms  works as a hack, but not the best option

**Comment by zombieJ** [comment]
Yes. `onSelect` trigger is before `onChange`. It's not good to trigger validate on `onSelect` since this should control by Form.
Add plus, validate function usually used on submit. Maybe you'd better to adjust the logic?
